resources/js/components/QuizzesPanel/QuizComponent.vue

```diff
@@ -0,0 +1,227 @@
+<script setup lang="ts">
+import { computed, onBeforeUnmount, ref } from 'vue'
+import dayjs, { type ConfigType as DayjsConfigType } from 'dayjs'
+import { type Quiz } from '@/Types/Quiz'
+import { type CleanQuestion } from '@/Types/CleanQuestion'
+import QuestionComponent from './QuestionComponent.vue'
+import EditableInput from '@/components/Common/EditableInput.vue'
+import ExapnsionToggleDynamicIcon from '@/components/Icons/ExapnsionToggleDynamicIcon.vue'
+import CopyIcon from '@/components/Icons/CopyIcon.vue'
+import LockIcon from '@/components/Icons/LockIcon.vue'
+import TrashIcon from '@/components/Icons/TrashIcon.vue'
+import PencilIcon from '../Icons/PencilIcon.vue'
+import CheckIcon from '../Icons/CheckIcon.vue'
+import DismissIcon from '../Icons/DismissIcon.vue'
+import { Request } from '@/scripts/request'
+import { nanoid } from 'nanoid'
+import '@vuepic/vue-datepicker/dist/main.css'
+import MessageBox, { useMessageBox } from '@/components/Common/MessageBox.vue'
+import CustomDatepicker from '../Common/CustomDatepicker.vue'
+import Banner from '../Common/Banner.vue'
+
+const props = defineProps<{quiz:Quiz, isSelected:boolean, showLockedQuizzes:boolean}>()
+const emit = defineEmits(['displayToggle'])
+const isEditing = ref(false)
+const quizRef = ref<Quiz>(props.quiz)
+quizRef.value.scheduledAt = formatDateHTML(quizRef.value.scheduledAt)
+const currentTime = ref(Date.now())
+const updateTimeInterval = setInterval(()=>currentTime.value = Date.now(), 1000)
+const confirmDeleteMessage = useMessageBox()
+onBeforeUnmount(()=>{clearInterval(updateTimeInterval)})
+
+const isReadyToSchedule = computed(()=>{
+  if (quizRef.value.scheduledAt && quizRef.value.duration)
+    return Date.parse(quizRef.value.scheduledAt) > currentTime.value
+  return false
+})
+
+// editing
+function edit(){
+  isEditing.value=true
+  if (!props.isSelected)
+    toggleQuizView()
+}
+function dismissEditing(){
+  quizRef.value = {...props.quiz}
+  isEditing.value=false
+}
+
+// date formatters
+function formatDatePretty(date?: DayjsConfigType) {
+  return date ? dayjs(date).format('DD.MM.YYYY HH:mm') : 'brak'
+}
+
+function formatDateHTML(date?: DayjsConfigType) {
+  return date ? dayjs(date).format('YYYY-MM-DDTHH:mm') : ''
+}
+
+function formatDateDB(date?: DayjsConfigType) {
```

Comment:
Where is function typing? What this function return ?

Reply:
![image](https://github.com/user-attachments/assets/0ff25345-c2f7-4b60-b2a1-488afd60bcd5)

Reply:
When this help you understand code without reading all function body. This is just example and I want to remind you can add return type in function, this can be useful.